src/core/cigar.rs

```diff
@@ -4,12 +4,13 @@ use crate::core::matter::{tables as matter, Matter};
 use crate::core::verfer::Verfer;
 use crate::error::{err, Error, Result};
 
+//  FIXME: writing uniffi bindings forced me to make fields public but it will be better to hid them
 #[derive(Debug, Clone, PartialEq)]
 pub struct Cigar {
-    raw: Vec<u8>,
-    code: String,
-    size: u32,
-    verfer: Verfer,
+    pub raw: Vec<u8>,
+    pub code: String,
+    pub size: u32,
+    pub verfer: Verfer,
```

Comment:
I don't know how I feel about this

Reply:
I do not like it either.  But when I write `uniffi` bindings for the library it spawns errors like this:
```
error[E0616]: field `verfer` of struct `cesride::Cigar` is private
   --> parside/uniffi/target/debug/build/parside-uniffi-837d61d2ba7e0a52/out/parside.uniffi.rs:383:69
    |
383 |         <FfiConverterTypeVerfer as uniffi::FfiConverter>::write(obj.r#verfer, buf);
```
I could not find a way to solve this error somehow else.

Reply:
Did you see this:
https://mozilla.github.io/uniffi-rs/udl/custom_types.html#using-custom-types-from-other-crates

Reply:
Unsure if it helps without playing with it

Reply:
Ah, I just saw your comment. I think the correct solution is probably something like:

```rust
pub struct Cigar {
    verfer: Verfer
}

impl Cigar {
    pub fn verfer() -> Verfer {
        <Self as Matter>::verfer()
    }
}
```

and then accessing through the public getter. We can wrap all the methods in the concrete classes that require it, it is just a bit more code. What do you think?

Reply:
In fact this particular getter exists, at least in my most recent branches

Reply:
Or is the error not on your access, but on binding generation?

Reply:
What I mean is - is it a result of your code or simply the integration with uniffi?

Reply:
yes - `binding generation`.  For access from parkside it's enough to have getters.
But when I try to generate bindings using `unifi` it threw a lot of errors as I mentioned above. 
And these errors are not only about `verfer` field (it was just an example), bout all fields (seems uniffi wants them all to be public).

Reply:
I see. Yeah it probably needs to construct the objects manually or something during conversions from the bound language. That's frustrating. Was the uniffi export thing of any use? Can we export in `cesride` and pass that uniffi export through to `parside`? I wasn't clear on how it worked

Reply:
This would actually provide a better user experience by allowing the choice between `parside` and `cesride` or just `cesride` if you don't need to parse.

Reply:
I can update my `cesride` uniffi PR and spike this out to see if it works